### PR TITLE
Fixed possible `NullPointerException` issue if health-check flow kick…

### PR DESCRIPTION
**Pull request by jeet1995**
In certain cases, if the channel health check flow kicks in before RNTBD context negotiation, the trying to obtain the RNTBD context in the health check flow can cause `NullPointerException`s to be thrown. This PR fixes such an issue to guarantee it to happen only after the channel has seen some requests allocated to it.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java [hunk 1]
```diff
@@ -426,6 +424,9 @@ private String isCancellationProneChannel(Timestamps timestamps, Instant current
             final long readSuccessRecency = Duration.between(timestamps.lastChannelReadTime(), currentTime).toNanos();
 
             if (readSuccessRecency >= this.nonRespondingChannelReadDelayTimeLimitInNanos) {
+
+                final Optional<RntbdContext> rntbdContext = requestManager.rntbdContext();
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java:428]: how moving it here helps?
`requestManager can still be null here, right?

**Review by kushagraThapar — CHANGES_REQUESTED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java (hunk 1 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java:428]: `requestManager` is checked for nullability at the beginning of `isHealthyWithFailureReason`. Moving it within at least guarantees the channel has seen wire reads which would imply RNTBD context negotiation has happened (in-line with other health check gates).

**Review by kushagraThapar — APPROVED** [review]

**Comment by jeet1995** [comment]
All live tests were passing for the commit - d621e47e001ebe8833f831d314d3ce6bc3693a38

**Pull request merged**